core/clipboard.ts

```diff
@@ -12,52 +12,92 @@ import {BlockPaster} from './clipboard/block_paster.js';
 import * as globalRegistry from './registry.js';
 import {WorkspaceSvg} from './workspace_svg.js';
 import * as registry from './clipboard/registry.js';
+import {Coordinate} from './utils/coordinate.js';
 
 /** Metadata about the object that is currently on the clipboard. */
-let copyData: ICopyData | null = null;
+let stashedCopyData: ICopyData | null = null;
 
-let source: WorkspaceSvg | null = null;
+let stashedWorkspace: WorkspaceSvg | null = null;
 
 /**
  * Copy a block or workspace comment onto the local clipboard.
  *
  * @param toCopy Block or Workspace Comment to be copied.
  * @internal
  */
-export function copy(toCopy: ICopyable) {
-  TEST_ONLY.copyInternal(toCopy);
+export function copy<T extends ICopyData>(toCopy: ICopyable<T>): T | null {
+  return TEST_ONLY.copyInternal(toCopy);
 }
 
 /**
  * Private version of copy for stubbing in tests.
  */
-function copyInternal(toCopy: ICopyable) {
-  copyData = toCopy.toCopyData();
-  source = (toCopy as any).workspace ?? null;
+function copyInternal<T extends ICopyData>(toCopy: ICopyable<T>): T | null {
+  const data = toCopy.toCopyData();
+  stashedCopyData = data; // Necessary for propery typing. This is why state sucks.
```

Comment:
Is this comment supposed to be about the line below it? I'm kinda confused about what you're saying is necessary for proper typing (?)

Reply:
Sorry this was confusing: We have to assign to `data` instead of assigning to `stashedCopyData` directly because `stashedCopyData` loses type info. I just removed the comment cuz it's kinda misleading :P